src/models/bugfindergame.ts

```diff
@@ -86,14 +32,40 @@ export class BugFinderGame {
    *
    * @throws {Error} when player already submitted bug for this code
    *
-   * @returns true when found right bug word and false when not
+   * @returns a list with mistakes in the code
    */
-  public submitWrongCode(bugWordId: number): boolean {
+  public submitWrongCode(submittedSolution: ISolution): Array<boolean> {
     if (this.hasSubmitted()) {
       throw Error('You already submitted this code.');
     }
-    this.solved[this.currentCodeNumber] = this.currentCode.bugWordId == bugWordId;
-    return this.solved[this.currentCodeNumber];
+    const playerBugs: IBug[] = submittedSolution.bugs;
+    const realSolution: ISolution = solutions[this.currentCodeNumber];
+    const bugs: IBug[] = realSolution.bugs;
+
+    const solutionFeedback: Array<boolean> = [];
+
+    bugs.forEach((bug) => {
+      const playerSpecificBug = playerBugs.find((playerBug) => playerBug.wordId == bug.wordId);
+
+      if (playerSpecificBug == null) {
+        solutionFeedback[bug.wordId] = false;
+      } else if (playerSpecificBug.errorType != bug.errorType) {
+        solutionFeedback[bug.wordId] = false;
+      } else {
+        solutionFeedback[bug.wordId] = true;
+      }
+    });
+
+    const falseSubmittedBugs = playerBugs.filter((playerBug) => {
+      return bugs.filter((bug) => bug.wordId != playerBug.wordId).length > bugs.length - 1;
```

Comment:
Is it really necessary that this line is that complicated?
Isn't there an easier way that doesn't require an inner loop and runs in constant time instead?

Reply:
we think yes. We struggled about 10 minutes with this line and this is the best solution we found. Maybe we will try to find a better solution again a later stage of this game. If you have a better solution we would be very happy

Reply:
```suggestion
      return bugs.filter((bug) => bug.wordId == playerBug.wordId).length == 0;
```
is at least clearer (and hopefully the correct logic).